js/src/wp-seo-dashboard-widget.js

```diff
@@ -0,0 +1,196 @@
+/* global wpseoDashboardWidgetL10n, wpseoApi, jQuery */
+
+import React from "react";
+import ReactDOM from "react-dom";
+
+import SeoAssessment from "yoast-components/composites/Plugin/DashboardWidget/components/SeoAssessment";
+import ScoreAssessments from "yoast-components/composites/Plugin/Shared/components/ScoreAssessments";
+import getFeed from "yoast-components/utils/getFeed";
+import WordpressFeed from "yoast-components/composites/Plugin/DashboardWidget/components/WordpressFeed";
+import colors from "yoast-components/style-guide/colors.json";
+
+class DashboardWidget extends React.Component {
+	/**
+	 * Creates the components and initializes its state.
+	 */
+	constructor() {
+		super();
+
+		this.state = {
+			statistics: null,
+			ryte: null,
+			feed: null,
+		};
+
+		this.getStatistics();
+		this.getRyte();
+		this.getFeed();
+	}
+
+	/**
+	 * Returns a color to be used for a given score.
+	 *
+	 * @param {string} score The score, expected to be 'na', 'bad', 'ok', 'good'.
+	 *
+	 * @returns {color} The color to use for this score. Defaults to grey if no such color exists.
+	 */
+	static getColorFromScore( score ) {
+		return colors[ `$color_${ score }` ] || colors.$color_grey;
+	}
+
+	/**
+	 * Fetches data from the statistics endpoint, parses it and sets it to the state.
+	 *
+	 * @returns {void}
+	 */
+	getStatistics() {
+		let that = this;
+
+		wpseoApi.get( "statistics", function( raw ) {
+			let statistics = {};
+
+			statistics.seoScores = raw.seo_scores.map( function( score ) {
+				return {
+					value: parseInt( score.count, 10 ),
+					color: DashboardWidget.getColorFromScore( score.seo_rank ),
+					html: `<a href="${ score.link }">${ score.label }</a>`,
+				};
+			} );
+
+			// Wrap in a div and get the text to HTML decode.
+			statistics.header = jQuery( `<div>${ raw.header }</div>` ).text();
+
+			that.setState( { statistics } );
+		} );
+	}
+
+	/**
+	 * Fetches data from the ryte endpoint, parses it and sets it to the state.
+	 *
+	 * @returns {void}
+	 */
+	getRyte() {
+		let that = this;
+
+		wpseoApi.get( "ryte", function( raw ) {
```

Comment:
`raw` feels like a non-descriptive parameter name. Please consider renaming this.